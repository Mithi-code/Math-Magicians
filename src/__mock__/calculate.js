const object =  {
  total: null,
  next: null,
  operation: null,
};

export default object
