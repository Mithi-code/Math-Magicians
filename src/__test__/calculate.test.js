import calculate from '../logic/calculate';
import obj from '../__mock__/calculate';

describe('Calculate', () => {
  test('the test is to return an empty object', () => {
    expect(calculate(obj, 'AC')).toEqual(obj);
    expect(calculate(obj, 'AC')).toMatchSnapshot();
  });

  test('the test should return 0', () => {
    obj.next = '0';
    expect(calculate(obj, '0')).toEqual({});
    expect(calculate(obj, '0')).toMatchSnapshot();
  });

  test('the test to should return concatenation of numbers with next value ', () => {
    obj.operation = '+';
    obj.next = '8';
    calculate(obj, '4');
    expect(calculate(obj, '4').next).toBe('84');
    expect(calculate(obj, '4').next).toMatchSnapshot();
  });

  test('the test should return a total null if  operation is null ', () => {
    obj.operation = null;
    obj.next = 2;
    expect(calculate(obj, '3').next).toBe('23');
    expect(calculate(obj, '3').total).toBe(null);
    expect(calculate(obj, '3').total).toMatchSnapshot();
  });

  test('the test should return a number entered if the operation and total are null', () => {
    obj.operation = null;
    obj.next = null;
    expect(calculate(obj, '3').next).toBe('3');
    expect(calculate(obj, '3').next).toMatchSnapshot();
  });
});