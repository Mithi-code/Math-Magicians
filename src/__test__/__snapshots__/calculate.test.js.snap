// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculate the test is to return an empty object 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`Calculate the test should return 0 1`] = `Object {}`;

exports[`Calculate the test should return a number entered if the operation and total are null 1`] = `"3"`;

exports[`Calculate the test should return a total null if  operation is null  1`] = `null`;

exports[`Calculate the test to should return concatenation of numbers with next value  1`] = `"84"`;
